bin/mutePublicSquadNotifications.ts

```diff
@@ -0,0 +1,68 @@
+import fastq from 'fastq';
+import '../src/config';
+import createOrGetConnection from '../src/db';
+import { Source, SourceMember } from '../src/entity';
+import {
+  NotificationPreferenceStatus,
+  NotificationType,
+  saveNotificationPreference,
+} from '../src/notifications/common';
+
+interface SourceMemberQueue {
+  userId: string;
+  sourceId: string;
+}
+
+const QUEUE_CONCURRENCY = 1;
+
+(async () => {
+  const con = await createOrGetConnection();
+  const stream = await con
+    .getRepository(SourceMember)
+    .createQueryBuilder('sm')
+    .select('sm."userId"')
+    .addSelect('sm."sourceId"')
+    .innerJoin(Source, 's', 's.id = sm."sourceId"')
+    .where('s.private IS FALSE')
+    .andWhere(
+      `
+      s.id NOT IN
+        (
+            SELECT  np."sourceId"
+            FROM    notification_preference np
+            WHERE   np."sourceId" = s.id
+            AND     np."referenceId" = s.id
+            AND     np."userId" = sm."userId"
+            AND     np."notificationType" = 'squad_post_added'
+            AND     np.status = 'muted'
```

Comment:
I think we should respect any already set notification preference.
We should only inject those who have it empty right?

Reasoning:
Someone might have muted, then turned back on.

Reply:
When they turn it back on, we remove the row, so we have no reference.

Reply:
Perhaps you can remove some criteria then?
I think the more you add, the more overhead right? and in those cases we wouldn't need for instance `referenceId` and `status`

Reply:
The criteria is to reduce the number of rows. If we remove the status we will pull everyone with existing muted status.

Reply:
I mean because of the np."notificationType" = 'squad_post_added' you only get those, which are only muted no?

Reply:
I do see it's only 653 rows so not too bad anyway :)

Reply:
We are fetching ones that are **not** muted. The total users to update is around 14k and seeing how the bin script performs, I am trying to lessen the number of rows as much as possible.